File: Backend/ThesisRestaurant/ThesisRestaurant.Contracts/Orders/CartResponse.cs
using ThesisRestaurant.Contracts.CustomFood;
using ThesisRestaurant.Contracts.Food;
using ThesisRestaurant.Contracts.FoodSize;
using ThesisRestaurant.Contracts.Ingredient;

namespace ThesisRestaurant.Contracts.Orders
{
    public record CartResponse(List<OrderCustomItemResponse> CustomFoods, List<OrderItemResponse> Foods);
    public record OrderCustomItemResponse(int Id, string Name, double Price, string FoodType, int Quantity, List<OrderItemIngredientResponse> Ingredients);
    public record OrderItemIngredientResponse(string Name, string IngredientType);
    public record OrderItemResponse(
        int Id, string Name, string FoodType, string FoodSize, int FoodSizeId, double FoodSizeMultiplier, List<OrderItemIngredientResponse> Ingredients,
        List<OrderAdditionalIngredientResponse> AdditionalIngredients, 
        int Quantity, double Price);

    public record OrderAdditionalIngredientResponse(int Id, string Name, string IngredientType, double Price, int Quantity);
}

File: Frontend/ThesisRestaurant/src/lib/types/classData/cartResponse.ts
export interface CartResponse {
    customFoods: Array<CartCustomFood>;
    foods: Array<CartFood>
}

export interface CartCustomFood {
    [key: string]: string | number | Array<CartIngredient>;
    guid: number;
    id: number;
    name: string;
    foodType: string;
    price: number;
    quantity: number;
    ingredients: Array<CartIngredient>
}

export interface CartFood {
    [key: string]: string | number | Array<CartIngredient> | Array<CartAdditionalIngredient>;
    guid: number;
    id: number;
    name: string;
    foodType: string;
    foodSizeId: number;
    foodSize: string;
    foodSizeMultiplier: number;
    price: number;
    quantity: number;
    ingredients: Array<CartIngredient>;
    additionalIngredients: Array<CartAdditionalIngredient>
}

export interface CartIngredient {
    name: string;
    ingredientType: string;
}

export interface CartAdditionalIngredient {
    id: number;
    name: string;
    ingredientType: string;
    price: number;
    quantity: number;
}